src/Symfony/Component/VarExporter/Hydrator.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarExporter;
+
+use Symfony\Component\VarExporter\Internal\Hydrator as InternalHydrator;
+
+/**
+ * A utility class to hydrate objects' properties.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+final class Hydrator
```

Comment:
Wouldn't it make sense to make this an own component? I mean (or I miss sth), doesn't this class do the opposite of var export? More a var "import" ? 

Thanks for explaining this to me, very interesting topic 😃

Reply:
Well, this class reuses all the internal APIs of VarExporter. The code _generated_ by VarExporter uses an hydrator because it has to recreate the object graph.
Moving this class to a different component would require duplicating the internal APIs or making them non-internal, which is a bad idea.

Reply:
Thanks